spec/lib/evss/disability_compensation_form/form526_to_lighthouse_transform_spec.rb

```diff
@@ -1,10 +1,53 @@
 # frozen_string_literal: true
 
 require 'rails_helper'
+require 'claims_api/form_schemas'
+require 'json_schema/json_api_missing_attribute'
 require 'evss/disability_compensation_form/form526_to_lighthouse_transform'
 
 RSpec.describe EVSS::DisabilityCompensationForm::Form526ToLighthouseTransform do
   let(:transformer) { subject }
+  let(:pdf_enabled_transformer) { EVSS::DisabilityCompensationForm::Form526ToLighthouseTransform.new(pdf_request: true) }
+  let(:pdf_disabled_transformer) { EVSS::DisabilityCompensationForm::Form526ToLighthouseTransform.new(pdf_request: false) }
+
+  def validate_lighthouse_schema(lh_request_body)
+    # This validation method mimics the same validation logic used in the actual controller
+    # https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/app/controllers/claims_api/v2/veterans/base.rb#L18
+    validator = ClaimsApi::FormSchemas.new(schema_version: 'v2')
+
+    # Convert the Vets::Model object to a hash and transform keys using the exact same
+    # transformation as the actual controller pipeline (OliveBranch with VA key patch)
+    request_body_hash = lh_request_body.as_json
+    camelized_request_body = transform_with_olivebranch(request_body_hash)
+
+    validator.validate!('GENERATE_PDF_526', camelized_request_body)
+  end
+
+  private
+
+  # Use the exact same transformation as the controller pipeline:
+  # OliveBranch transformation with VA key patch applied using the actual middleware logic
+  def transform_with_olivebranch(data)
+    # Transform using OliveBranch (same as the actual controller)
+    camelized = OliveBranch::Transformations.transform(
+      data,
+      OliveBranch::Transformations.method(:camelize)
+    )
+
+    # Apply the VA key patch using the same regex and logic as the middleware
+    json_string = camelized.to_json
+    # Use the same VA_KEY_REGEX pattern and gsub logic as the middleware extension
+    json_string.gsub!(/("[^"]+VA[^"]*"):/) do
+      key = Regexp.last_match(1)
```

Comment:
TODO: check if this needs to be a global replace. Why is it just doing it on the last match?